### PR TITLE
Revert Enum default computation

### PR DESCRIPTION
**Pull request by corranwebster**
Revert enum defaults to using index 0 except for `enum.Enum`s, which instead use the first value returned from the iterator.

This restores the default behaviour to exactly what it was before the introduction of support for `enum.Enum` other than special-casing default values for `enum.Enum`.

### CODE REVIEW

File: traits/trait_base.py [hunk 1]
```diff
@@ -211,20 +211,27 @@ def safe_contains(value, container):
...
 
     Returns
     -------
     default : any
-        The first item of the collection, or None if the collection is empty.
+        The first valid value, or None if the collection is empty.
     """
-    return next(iter(collection), None)
+    if isinstance(values, enum.EnumMeta):
+        # XXX special-case enum.Enums; generalize to Collections?
```
Comment (mdickinson) [inline traits/trait_base.py]: Can we turn this into an issue and get rid of the XXX comment?

File: traits/trait_base.py [hunk 2]
```diff
@@ -211,20 +211,27 @@ def safe_contains(value, container):
         return False
 
 
-def collection_default(collection):
-    """ Get the first item of a collection, returning None if empty.
+def enum_default(values):
+    """ Get the first item of an EnumType, returning None if empty.
```
Comment (mdickinson) [inline traits/trait_base.py]: Replace "EnumType" with "Enum subclass"? (It's not clear to me what `EnumType` refers to here.)

Comment (mdickinson) [reply traits/trait_base.py]: Ah, I see; `EnumType` means any of the types listed in `EnumTypes`. Maybe we can just rephrase in plain English?

**Comment by corranwebster** [comment]
See #835 for a full proposal.

**Comment by kitchoi** [comment]
> Does this seem like a satisfactory resolution to #812 to you?

Yes it does!  Thanks both!
